woocommerce/woocommerce-ios
# Dynamic Dashboard: Load data for stock card

Closes: #12796 
Closes #12754 


## Description

This PR completes the functionality for the stock dashboard card with logic and content on the UI. Changes include:
- Updated the view model to load data. The logic is to load stock report based on the selected stock type, then either fetch product reports & product details or use in-memory data if already loaded before.
- Mapped loaded details to a new model `StockItem` to display on the stock card UI.
- Handled empty state and error states for the card.
- Handled navigation from the stock item to open the detail screen for the relevant product.

## Testing instructions

- Log in to the app and enable the Stock card.
- Switch between different stock type and confirm that the card displays the correct items of the selected status with the correct stock quantity and number of items sold in the last 30 days. To confirm the data, you can use Core's Analytics to cross-check.
- Tap on any stock item, confirm that the detail form is displayed for that product.
- Go to wp-admin > WooCommerce > Settings > Advanced > Features > disable Woo Analytics and reload the app.
- Confirm that the error state for disabled Analytics is displayed.
- Enable Analytics again, then disconnect the Internet before reloading. Confirm that the generic error state screen is displayed.
- Enable the connection then switch to a store with no stock of a certain status. Confirm that the empty state is displayed when that status is selected.

## Screenshots

Empty state | Analytics disabled | Generic error | With data
--- | --- | --- | ---
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/b138237a-978c-4925-b87c-fad1243398c4" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/bc379df9-0718-4f6a-b50d-f4b51a4ac44c" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/ddd32652-7a6a-4468-bed5-23c73caba487" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/7151d8b8-3d87-4132-af36-398ba7cb9f7d" width=320 /> 



---
